enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx

```diff
@@ -47,29 +54,45 @@ export const _InteractiveQuestion = ({
   const card = useSelector(getCard);
   const result = useSelector(getFirstQueryResult);
   const uiControls = useSelector(getUiControls);
-  const queryResults = useSelector(getQueryResults);
+  const hasQuestionChanges =
+    card && (!card.id || card.id !== card.original_card_id);
 
   const [loading, setLoading] = useState(true);
 
   const { isRunning } = uiControls;
 
-  useEffect(() => {
-    const fetchQBData = async () => {
-      const { location, params } = getQuestionParameters(questionId);
-      dispatch(initializeQB(location, params));
-    };
+  if (question) {
+    // FIXME: remove "You can also get an alert when there are some results." feature for question
+    question.alertType = returnNull;
+  }
 
-    fetchQBData().then(() => setLoading(false));
+  const loadQuestion = async (
+    dispatch: ReturnType<typeof useDispatch>,
```

Comment:
Looking at [the react-redux docs](https://react-redux.js.org/api/hooks#usedispatch), `dispatch` here would always be stable and unchanging, so I think we don't need to pass in dispatch here.

> The dispatch function reference will be stable as long as the same store instance is being passed to the Provider. Normally, that store instance never changes in an application.